sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/MediaJobErrorCategoryExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Messaging.EventGrid.SystemEvents
+{
+    internal static partial class MediaJobErrorCategoryExtensions
+    {
+        public static MediaJobErrorCategory ToMediaJobErrorCategory(this string value)
+        {
+            if (string.Equals(value, "Service", StringComparison.InvariantCultureIgnoreCase)) return MediaJobErrorCategory.Service;
```

Comment:
Would it be worth doing a lower-case transform on `value` once and then using `StrimgComparison.InvariantCulture` rather than potentially running multiple case insensitive comparisons?

Reply:
I don't think so. This is an exact copy of the generated code, other than Int32.MaxValue part.